File: algorithms/math/factorial.test.dart
import 'factorial.dart';

testFactorial() {
  var num = factorial(0), expected = 1;
  if (num != expected) {
    print('expect factorial(0) to be: $expected, got instead: $num');
  }

  num = factorial(1); expected = 1;
  if (num != expected) {
    print('expect factorial(1) to be: $expected, got instead: $num');
  }

  num = factorial(2); expected = 2;
  if (num != expected) {
    print('expect factorial(2) to be: $expected, got instead: $num');
  }

  num = factorial(6); expected = 720;
  if (num != expected) {
    print('expect factorial(6) to be: $expected, got instead: $num');
  }
}

main() {
  testFactorial();
}

File: algorithms/math/factorial.test.ts
import factorial from "./factorial.ts";

function testFactorial() {
  let num = factorial(0), expected = 1;
  console.assert(
    num === expected,
    `expect factorial(0) to be: ${expected}, got instead: ${num}`,
  );

  num = factorial(1), expected = 1;
  console.assert(
    num === expected,
    `expect factorial(1) to be: ${expected}, got instead: ${num}`,
  );

  num = factorial(2), expected = 2;
  console.assert(
    num === expected,
    `expect factorial(2) to be: ${expected}, got instead: ${num}`,
  );

  num = factorial(6), expected = 720;
  console.assert(
    num === expected,
    `expect factorial(6) to be: ${expected}, got instead: ${num}`,
  );
}

testFactorial();
